### PR TITLE
Add label-routes and controllers

### PR DESCRIPTION
**Pull request by carpecodeum**
# Description
Add routes and controllers for labels

Fixes #424 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: labellab-flask/api/controllers/labelscontroller.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                                     force=True)
+        try:
+            label_name = post_data["label_name"]
+            label_type = post_data["label_type"]
+        except Exception:
+            response = {
+                "success": False,
+                "msg": "Please provide all the required fields."
+            }
+            return make_response(jsonify(response)), 500
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:38]: Status should be 400

File: labellab-flask/api/controllers/labelscontroller.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        """
+        Handle POST request for this view.
+        Url --> /api/v1/label/create/<int:project_id>
+        """
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        try:
+            label_name = post_data["label_name"]
+            label_type = post_data["label_type"]
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:32]: Should validate the content of ```label_type```. In my opinion, let's keep ```rectangle``` and ```polygon``` as supported types and update when we implement other types of labels. @carpecodeum what's your idea?

File: labellab-flask/api/controllers/labelscontroller.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        Handle GET request for this view. 
+        Url ---> /api/v1/label/get/<int:project_id>
+        """
+        try:
+            if not project_id:
+                response = {
+                    "success": False,
+                    "msg": "Provide the project_id."
+                }
+                return make_response(jsonify(response)), 500
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:80]: Status should be 400

File: labellab-flask/api/controllers/labelscontroller.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    """
+    @jwt_required
+    def get(self, label_id):
+        try:
+            if not label_id:
+                response = {
+                    "success":False,
+                    "msg": "Label id not provided"
+                }
+                return make_response(jsonify(response)), 500
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:123]: Status should be 400

File: labellab-flask/api/controllers/labelscontroller.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+    @jwt_required
+    def delete(self, label_id):
+        try:
+            if not label_id:
+                response = {
+                    "success":False,
+                    "msg": "Label id not provided"
+                    }
+                return make_response(jsonify(response)), 500
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:150]: Status should be 400

File: labellab-flask/api/controllers/labelscontroller.py [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+    @jwt_required
+    def put(self, label_id):
+        """Handle PUT request for this view. Url --> /api/v1/Label/update"""
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        try:
+            label_name = post_data["label_name"]
+            label_type = post_data["label_type"]
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:176]: Should validate ```label_type``` here too.

File: labellab-flask/api/controllers/labelscontroller.py [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        except Exception:
+            response = {
+                "success": False,
+                "msg": "Please provide all the required fields."}
+            return make_response(jsonify(response)), 404
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/labelscontroller.py:181]: Status should be 400

**Comment by carpecodeum** [comment]
hey I am facing some problems with my laptop and hopefully fix those problems and these issues soon. I am sorry for the delay @UdeshUK @ivantha

**Pull request merged**